src/main/java/bitflow4j/script/Main.java

```diff
@@ -106,20 +104,54 @@ private static void logCapabilities(Registry registry) {
     private static void logIfNotEmpty(String title, Collection<? extends RegisteredStep> reg) {
         if (reg.isEmpty())
             return;
-        logger.info(""); // Force new line in log output
-        logger.info(title);
+        System.out.println(); // Fore new line
+        System.out.println(title);
         reg.stream().sorted(stepComparator).forEach(Main::logCapability);
     }
 
     private static void logCapability(RegisteredStep registeredPipelineStep) {
-        logger.info(" - " + registeredPipelineStep.getStepName());
-        logger.info("     Description: " + registeredPipelineStep.getDescription());
+        System.out.println(" - " + registeredPipelineStep.getStepName());
+        System.out.println("     Description: " + registeredPipelineStep.getDescription());
         if (!registeredPipelineStep.getRequiredParameters().isEmpty())
-            logger.info("     Required parameters: " + registeredPipelineStep.getRequiredParameters());
+            System.out.println("     Required parameters: " + registeredPipelineStep.getRequiredParameters());
         if (!registeredPipelineStep.getOptionalParameters().isEmpty())
-            logger.info("     Optional parameters: " + registeredPipelineStep.getOptionalParameters());
+            System.out.println("     Optional parameters: " + registeredPipelineStep.getOptionalParameters());
         if (registeredPipelineStep.hasGenericConstructor())
-            logger.info("     Accepts any parameters");
+            System.out.println("     Accepts any parameters");
+    }
+
+    private static void logJsonCapabilities(Registry registry) {
+        Map<String, Object> root = new HashMap<>();
+        addJsonCapabilities(root, "stream_processing_steps", registry.getStreamCapabilities());
+        addJsonCapabilities(root, "batch_processing_steps", registry.getBatchCapabilities());
+        addJsonCapabilities(root, "fork_steps", registry.getForkCapabilities());
+        System.out.println(new Gson().toJson(root));
```

Comment:
not sure if the object type can be mapped automatically by gson to json

Reply:
There is a test that checks that this method produces correct (parsable) JSON, and that it contains certain keywords. I have not seen the JSON output myself, yet...